src/platform.ts

```diff
@@ -79,6 +79,18 @@ export class MideaPlatform implements DynamicPlatformPlugin {
    */
   private finishedLaunching() {
     this.log.info('Start device discovery...');
+    // If IP address is in config then probe them directly
+    this.platformConfig.devices.forEach((device) => {
+      // for some reason, assigning the regex has to be inside the loop, else fails after first pass.
+      const regexIPv4 = /^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$/gi;
+      const ip = device.advanced_options?.ip?.toString().trim();
```

Comment:
I don't know how the TypeScript compiler handles this as int the DeviceConfig type advanced_options and ip is not an optional field.

Reply:
I forgot that you proposed adding in a merge with defaults.  I have put that in and removed the question marks.

Reply:
Maybe don't need to merge the whole defaultDeviceConfig. It's enough to define only the IP field. So like 

```js
device = {
   ...device,
   advanced_options: {
      ip: '',
      ...device.advanced_options
   }
}
```

Reply:
That feels more complicated than just using question marks ?  Are you sure this doesn't overwrite existing value for ip?  Is it order dependent, so existing advanced options overwrite `ip: '',` I'm just not very familiar with this construct.

Reply:
It's just weird for me to use question marks here when in theory those fields cannot be empty. The order I wrote is the correct one, but you can test it in Node in your terminal.

Reply:
Yeah, basically what's later in the code will overwrite previous fields. So in this case, if the device advanced_options has the ip field in it, it will overwrite the empty one. Maybe it also be handled if the advanced_options is totally empty.

Reply:
Hummm...
<img width="879" alt="image" src="https://github.com/kovapatrik/homebridge-midea-platform/assets/17837612/d8c17bf9-edbf-4a73-8014-53c774cfac61">

Reply:
Yeah this error is exactly why the question mark handling is weird for me: TypeScript sees the advanced_options field to always have values, because the type describe that.

Reply:
So overcome this error, the field ip should be assigned like this

```js
device = {
   ...device,
   advanced_options: {
      ip: device.advanced_options && device.advanced_options.ip ?  device.advanced_options.ip : '' 
      ...device.advenced_options
   }
}
```

Reply:
Or just simply

```js
device.advanced_options.ip = device.advanced_options.ip ?? '';
```

Reply:
But again, this check is kinda weird, but I would prefer this format against the question mark handling.

Reply:
You still have to handle advanced_options being undefined.  If you don't like the question marks then I think the safest thing to do is merge with defaults, which is what we have now.